bigchaindb_driver/pool.py

```diff
@@ -63,7 +75,45 @@ def __init__(self, connections, picker_class=RoundRobinPicker):
 
         """
         self.connections = connections
+        self.node_count = len(self.connections)
+        self.max_retries = self.node_count * 4
+        self.retries = {
+            key: self.max_retries for key in range(self.node_count)
+        }
         self.picker = picker_class()
+        self.DELAY = 60
+
+    def retries_left(self, node, success=True):
+        """Update retries left of the current node"""
+        if success:
+            self.retries[node] = min(self.retries[node] + 1, self.max_retries)
+        else:
+            self.retries[node] = max(self.retries[node] - 1, 0)
+
+    def fail_node(self):
+        """Send a message to the pool indicating the connection
+        to the current node is failing and needs to try another one
+        """
+        failing_node = self.picker.picked
+        self.retries_left(failing_node, success=False)
+        self.connections[failing_node]["time"] = datetime.now(
+        ) + timedelta(seconds=self.DELAY)
+        self.picker.next_node(self.connections)
+
+    def nodes_available(self):
+        """Check if there are retries left in every node"""
+        for node in self.retries:
+            if self.retries[node] > 0:
+                return True
+        return False
+
+    def success_node(self):
+        """Send a message to the pool indicating the connection to the current
+        node is succesful
+        """
+        success_node = self.picker.picked
+        self.retries_left(success_node, success=True)
+        self.picker.next_node(self.connections)
```

Comment:
I think, we should stay to exploit the node if the request is successful. If the node is capable of serving so many requests the client request times will be improved significantly. If the node can't handle it, we back off. Also note that each driver user assembles the list of the nodes by himself so the initial distribution should be close to uniform.